spotirec/oauth2.py

```diff
@@ -10,15 +10,15 @@
 class SpotifyOAuth:
     OAUTH_AUTH_URL = 'https://accounts.spotify.com/authorize'
     OAUTH_TOKEN_URL = 'https://accounts.spotify.com/api/token'
-    PORT = 8080
+    PORT = 0
     LOGGER = None
     CONF = None
     API = None
 
     def __init__(self):
         self.client_id = '466a89a53359403b82df7d714030ec5f'
         self.client_secret = '28147de72c3549e98b1e790f3d080b85'
-        self.redirect = f'http://localhost:{self.PORT}'
+        self.redirect = f'http://localhost'
```

Comment:
Have you tried to not specify the protocol and see if it still works? I could imagine some concoction of extensions and HTTPSEverywhere could bork this.

Reply:
Through discussion, we've decided that this testing is out of scope - maybe issue-worthy but I can't be bothered.